src/linter.cc

```diff
@@ -22,92 +22,245 @@
+    // outside of english lowercase letters.
+    for ( auto &token : parse_tokens ) {
+      if ( token.kind() == ParseToken::KEYWORD ) {
+        if (!AllUpperCase(sql, token.GetLocationRange())) {
+          return absl::Status(
+            absl::StatusCode::kFailedPrecondition,
+            absl::StrCat("All keywords should be Uppercase, In character ",
+            std::to_string(token.GetLocationRange().start().GetByteOffset()),
+            " string should be: ",
+            token.GetSQL()));
+        }
+      }
+    }
+
+    return absl::OkStatus();
+}
+
+absl::Status CheckCommentType(absl::string_view sql) {
+    bool includes_type1 = false;
+    bool includes_type2 = false;
+    bool includes_type3 = false;
+    bool inside_string = false;
 
-    std::unique_ptr<zetasql::ParserOutput> output;
+    for (int i = 1; i<static_cast<int>(sql.size()); ++i) {
+        if (!inside_string && sql[i-1] == '-' && sql[i] == '-') {
+            includes_type1 = true;
+            // ignore the line.
+            while ( i < static_cast<int>(sql.size()) &&
+                    sql[i] != '\n' ) {
+                ++i;
+            }
+        }
 
-    absl::Status parser_status = zetasql::ParseStatement(sql,
-            zetasql::ParserOptions(), &output);
+        if (!inside_string && sql[i-1] == '/' && sql[i] == '/') {
+            includes_type2 = true;
+            // ignore the line.
+            while ( i < static_cast<int>(sql.size()) &&
+                    sql[i] != '\n' ) {
+                ++i;
+            }
+        }
 
-    if ( !parser_status.ok() )
-        return parser_status;
-    // TODO(orhan_uysal): Implement KeywordExtractor.
+        if (!inside_string && sql[i] == '#') {
+            includes_type3 = true;
+            // ignore the line.
+            while ( i < static_cast<int>(sql.size()) &&
+                    sql[i] != '\n' ) {
+                ++i;
+            }
+        }
 
-    for (const zetasql::ASTNode *keyword : keywords) {
-        if ( !allUpperCase(keyword) ) {
-            return absl::Status(
-                absl::StatusCode::kFailedPrecondition, "");
+        // ignore multiline comments.
+        if (!inside_string && sql[i-1] == '/' && sql[i] == '*') {
+            // it will start checking after '/*' and after the iteration
+            // finished, the pointer 'i' will be just after '*/' (incrementation
+            // from the for statement is included).
+            i += 2;
+            while ( i < static_cast<int>(sql.size()) &&
+                    !(sql[i-1] == '*' && sql[i] == '/') ) {
+                ++i;
+            }
         }
+
+        if (sql[i] == '\'' || sql[i] == '"')
+            inside_string = !inside_string;
     }
 
+    if ( includes_type1 + includes_type2 + includes_type3 > 1 )
+        return absl::Status(
+                absl::StatusCode::kFailedPrecondition,
+                "either '//' or '--' should be used to "
```

Comment:
You'd need to construct here more specific error message, e.g.:
Different comment types used in file:
line <x> has 'type1'
line <y> has 'type2'

Again, can be fixed in a follow up commit.

Reply:
I will fix it it next PR.